R/data_reshape.R

```diff
@@ -175,83 +175,92 @@ data_to_long <- function(data,
 #' @export
 data_to_wide <- function(data,
                          values_from = "Value",
-                         colnames_from = "Name",
-                         sep = "_",
-                         ...,
-                         names_from = colnames_from,
+                         names_from = "Name",
+                         names_sep = "_",
+                         values_fill = NULL,
                          verbose = TRUE,
-                         values_fill = NULL) {
+                         ...,
+                         colnames_from,
+                         sep) {
+
+  .is_deprecated(colnames_from, "names_from")
```

Comment:
Why don't you just pass the old argument name as character/string as well?

Reply:
because `colnames_from` needs to be evaluated in `missing()` in `.is_deprecated` but it turns out this function doesn't work correctly when using the pipe. 
```r
> fish_encounters %>%
+     reshape_wider(
+       colnames_from = "station",
+       values_from = "seen",
+       names_prefix = "foo_"
+     )
# A tibble: 114 × 3
   fish  station foo_NA
 * <fct> <fct>    <int>
 1 4842  Release      1
 2 4842  I80_1        1
 3 4842  Lisbon       1
 4 4842  Rstr         1
 5 4842  Base_TD      1
 6 4842  BCE          1
 7 4842  BCW          1
 8 4842  BCE2         1
 9 4842  BCW2         1
10 4842  MAE          1
# … with 104 more rows
Warning message:
Argument `.` is deprecated. Please use `names_from` instead. 
```
Do you have a better idea of a function that automatically checks if a deprecated arg is called and shows a message if it is?

Reply:
not sure if this works in pipes, but maybe:

```r
if (!missing(colnames_from)) {
  .is_deprecated("colnames_from", "names_from") # simplify `.is_deprecated()`
  if (is.null(names_from)) {
    names_from <- colnames_from
  }
}

Reply:
```r
.is_deprecated <- function(arg, replacement) {
  warning(insight::format_message(
    paste0("Argument `", arg, "` is deprecated. Please use `", replacement, "` instead.")
  ), call. = FALSE)
}
```

Reply:
This will probably work but the idea was that the check with `missing` was made inside `.is_deprecated` to remove all the if conditions in the main functions

Reply:
I applied your method, it's better